### PR TITLE
[Core]: Add virtual CAN driver plugin with unit tests

### PR DESCRIPTION
**Pull request by GwnDaan**
### What's new:
- Added VirtualCANPlugin to communicate with ourself in testing environments.
- Some basic unit tests for the VirtualCANPlugin.
- Modified building of GTest such that it will still compile the tests that don't rely on SocketCAN when SocketCAN is not available.
- Moved setting of `--coverage` flag to CXX compiler up so SonerCloud will hopefully now track the other files for code coverage as well.

TLDR: I just wrote my first C++ unit test 😄

### CODE REVIEW

**Comment by ad3154** [comment]
This is great! Now we can replace the SocketCAN in the address claim test, and do much _much_ more testing than before!

File: hardware_integration/src/virtual_can_plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
+#include "isobus/hardware_integration/virtual_can_plugin.hpp"
```
Comment (ad3154) [inline hardware_integration/src/virtual_can_plugin.cpp:1]: Need comment block at top of file

File: test/virtual_can_plugin_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	EXPECT_EQ(receiveFrame.data[0], 0x01);
+	EXPECT_EQ(receiveFrame.data[1], 0x02);
+	EXPECT_EQ(receiveFrame.data[2], 0x03);
+	EXPECT_EQ(receiveFrame.data[3], 0x04);
+	EXPECT_EQ(receiveFrame.data[4], 0x05);
+	EXPECT_EQ(receiveFrame.data[5], 0x06);
+	EXPECT_EQ(receiveFrame.data[6], 0x07);
+	EXPECT_EQ(receiveFrame.data[7], 0x08);
+	EXPECT_EQ(receiveFrame.dataLength, 8);
+}
```
Comment (ad3154) [inline test/virtual_can_plugin_tests.cpp:72]: Add newline at the end of the file

**Comment by ad3154** [comment]
In theory with this we should be able to fully simulate a TP and ETP session, test all abort conditions, etc, so that's very cool

**Comment by ad3154** [comment]
I wish I had infinite time to work on this, haha

**Comment by GwnDaan** [comment]
> I wish I had infinite time to work on this, haha

Same haha, I'm mostly postponing my other projects now till the vt client can resize the object pools, hence I have more time to collaborate on this stack. And it helps that it's winter so not much work needs to be done on the farm.

File: hardware_integration/include/isobus/hardware_integration/virtual_can_plugin.hpp [hunk 3]
```diff
@@ -0,0 +1,85 @@
+//================================================================================================
+/// @file virtual_can_plugin.hpp
+///
+/// @brief An OS and hardware independent virtual CAN interface driver for testing purposes.
+/// Any instance connecting to the same channel and in the same process will be able to communicate.
+/// @author Daan Steenbergen
+///
+/// @copyright 2023 Adrian Del Grosso
+//================================================================================================
+#ifndef SOCKET_CAN_INTERFACE_HPP
```
Comment (ad3154) [inline hardware_integration/include/isobus/hardware_integration/virtual_can_plugin.hpp:10]: Probably need to change the include guards from `SOCKET_CAN_INTERFACE_HPP` to `VIRTUAL_CAN_PLUGIN_HPP`

File: hardware_integration/include/isobus/hardware_integration/virtual_can_plugin.hpp [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+	const std::string channel; ///< The virtual channel name
+	const bool receiveOwnMessages; ///< If `true`, the driver will receive its own messages
+
+	std::condition_variable condition; ///< A condition variable to wake us up when a frame is received
+	VirtualDevice *ourDevice; ///< A pointer to the virtual device of this instance
+	bool running; ///< If `true`, the driver is running
+};
+
+#endif // SOCKET_CAN_INTERFACE_HPP
```
Comment (ad3154) [inline hardware_integration/include/isobus/hardware_integration/virtual_can_plugin.hpp:85]:
```suggestion
#endif // VIRTUAL_CAN_PLUGIN_HPP

```

**Review by ad3154 — CHANGES_REQUESTED** [review]

File: hardware_integration/include/isobus/hardware_integration/virtual_can_plugin.hpp (hunk 3 above)
Comment (GwnDaan) [reply hardware_integration/include/isobus/hardware_integration/virtual_can_plugin.hpp]: My bad haha, I changed it

**Review by ad3154 — APPROVED** [review]
